### PR TITLE
bug: issue fixer creates PRs with invalid commit messages for kubevirt repos

### LINKED ISSUE #229: bug: issue fixer creates PRs with invalid commit messages for kubevirt repos
## Summary

When oompa's issue fixer creates PRs for kubevirt org repos, the commit message format violates kubevirt's commit message standards, causing prow's `invalid-commit-message` check to fail.

## Example

PR [kubevirt/cluster-network-addons-operator#2800](https://github.com/kubevirt/cluster-network-addons-operator/pull/2800) has this commit message:

```
Fix #2799: CI Failure: pull-e2e-cluster-network-addons-operator-monitoring-k8s / The `pull-e2e-cluster-network-addons-operator-monitoring-k8s
```

Prow flagged it as invalid:
> Keywords which can automatically close issues and hashtag(#) mentions are not allowed.

## Problems

1. **`Fix #NNN:` prefix** — kubevirt prow rejects `Fix`, `Fixes`, `Closes` keywords that auto-close issues
2. **Title too long** — the auto-generated title includes the full issue title (job name + failure description), making the first line far over 72 characters
3. **Issue title leaked into commit subject** — the commit subject should be a concise description of the change, not a copy of the issue title

## Expected behavior

Oompa should generate commit messages that follow the target repo's conventions. For kubevirt repos:
- No auto-close keywords (`Fix`, `Fixes`, `Closes`) in the commit message
- First line under 72 characters
- Reference the issue in the body (e.g., `Related-to: #2799`) rather than the subject line

### PR DESCRIPTION
**Pull request by qinqon**
Fixes #229

## Summary

Fix commit message format to avoid auto-close keywords and enforce 72-character subject lines, preventing kubevirt prow's `invalid-commit-message` check from rejecting oompa-created PRs.

## Changes

- Replace `Fix #N: <title>` and `Fixes #N` commit subject formats with the issue title directly and `Related-to: #N` in the body
- Add `truncateSubject()` helper that truncates at word boundaries with `...` suffix when the subject exceeds 72 characters
- Update all existing commit message tests to verify the new format
- Add new test for long issue titles verifying truncation behavior
- Add unit tests for `truncateSubject` edge cases

## Testing

- [x] `go test ./...` passes
- [x] `go vet ./...` passes
- [x] Manual testing (if applicable)

## Related Issues

Fixes #229

### CODE REVIEW

File: pkg/agent/git_ops.go [hunk 1]
```diff
@@ -117,6 +117,28 @@ func isConventionalCommitTitle(title string) bool {
...
+	cutoff := maxLen - 3
+	if cutoff <= 0 {
+		return subject[:maxLen]
+	}
+	// Try to break at the last space before the cutoff.
+	if idx := strings.LastIndex(subject[:cutoff], " "); idx > 0 {
+		return subject[:idx] + "..."
+	}
+	return subject[:cutoff] + "..."
+}
```
[the remark this answers is not in the document]
Comment (qinqon) [reply pkg/agent/git_ops.go:143]: Fixed. Converted `truncateSubject` to operate on `[]rune` instead of byte indices, so multi-byte UTF-8 characters (emojis, accented characters, CJK, Cyrillic, etc.) are never split mid-sequence. The word-boundary logic is preserved — the function still breaks at the last space before the rune-count limit. Added a multi-byte test case (`"Ошибка CI: ..."`) to verify correct behavior with Cyrillic characters.

Note: I didn't directly reuse `truncateRunes` from `cmd/oompa/tui.go` because it's in a different package and lacks the word-boundary-aware truncation that commit subjects need. Instead, I made `truncateSubject` itself rune-safe.

[the remark this answers is not in the document]
Comment (qinqon) [reply pkg/agent/git_ops.go:143]: Fixed. Same issue — converted `truncateSubject` to use `[]rune` for all length checks and slicing operations, so multi-byte UTF-8 characters are handled correctly. Added a test case with Cyrillic characters to verify.

**Comment by qinqon** [comment]
[Change](https://github.com/qinqon/oompa/compare/16f507a34703dbc7ad89e193ee8354a80142db49..3322a4d1c7b8e2d261e5b2dc031da5f49655ea5f)
- `pkg/agent/git_ops.go` (15 +++++++++------)
- `pkg/agent/loop_test.go` (5 +++--)
